pydicom/multival.py

```diff
@@ -30,96 +30,110 @@ class MultiValue(MutableSequence):
 
-    @overload
-    def __setitem__(self, i: int, val: object) -> None: pass
-
-    @overload
-    def __setitem__(self, i: slice, val: Sequence[object]) -> None: pass
-
-    def __setitem__(
-        self, i: Union[slice, int], val: Union[Sequence[object], object]
-    ) -> None:
-        """Set an item of the list, making sure it is of the right VR type"""
-        if isinstance(i, slice):
-            val = cast(Sequence[object], val)
-            val = [self.type_constructor(v) for v in val]
-            self._list.__setitem__(i, val)
-        else:
-            self._list.__setitem__(i, self.type_constructor(val))
+    def __delitem__(self, index: Union[slice, int]) -> None:
+        del self._list[index]
 
-    def __str__(self) -> str:
-        if not self:
-            return ''
-        lines = [
-            f"'{x}'" if isinstance(x, (str, bytes)) else str(x) for x in self
-        ]
-        return f"[{', '.join(lines)}]"
+    def extend(self, val: Iterable[_T]) -> None:
+        """Extend a :class:`~pydicom.dataset.Dataset` to the sequence."""
+        self._list.extend([self.type_constructor(x) for x in val])
 
-    __repr__ = __str__
+    def __iadd__(self, other: Iterable[_T]) -> MutableSequence[_ItemType]:
+        """Implement MultiValue() += [object]."""
+        self._list += [self.type_constructor(x) for x in other]
+        return self
 
-    def __len__(self) -> int:
-        return len(self._list)
+    def __eq__(self, other: object) -> bool:
+        return self._list == other
 
     @overload
-    def __getitem__(self, index: int) -> object: pass
+    def __getitem__(self, index: int) -> _ItemType: pass
 
     @overload
-    def __getitem__(self, index: slice) -> Sequence[object]: pass
+    def __getitem__(self, index: slice) -> MutableSequence[_ItemType]: pass
 
     def __getitem__(
         self, index: Union[slice, int]
-    ) -> Union[Sequence[object], object]:
+    ) -> Union[MutableSequence[_ItemType], _ItemType]:
         return self._list[index]
 
-    def __delitem__(self, index: Union[slice, int]) -> None:
-        del self._list[index]
+    def insert(self, position: int, val: _T) -> None:
+        self._list.insert(position, self.type_constructor(val))
 
-    def __iter__(self) -> Iterator[object]:
+    def __iter__(self) -> Iterator[_ItemType]:
         return iter(self._list)
 
-    def __eq__(self, other: object) -> bool:
-        return self._list == other
+    def __len__(self) -> int:
+        return len(self._list)
 
     def __ne__(self, other: object) -> bool:
         return self._list != other
 
+    @overload
+    def __setitem__(self, idx: int, val: _T) -> None: pass
+
+    @overload
+    def __setitem__(self, idx: slice, val: Iterable[_T]) -> None: pass
+
+    def __setitem__(
+        self, idx: Union[slice, int], val: Union[Iterable[_T], _T]
+    ) -> None:
+        """Set an item of the list, making sure it is of the right VR type"""
+        if isinstance(idx, slice):
```

Comment:
Stuck on the mypy warning for `isinstance(idx, slice)`, can't figure out what I should be doing...

> Expression type contains 'Any' (has type Type[slice])